src/parser/core/modules/DoTs.js

```diff
@@ -75,12 +75,20 @@ export default class DoTs extends Module {
 		// Remove any untargetable time from the clip - often want to hardcast after an invuln phase, but refresh w/ 3D shortly after.
 		clip -= this.invuln.getUntargetableUptime('all', event.timestamp - this._statusDuration[statusId], event.timestamp)
 
-		// Also remove invuln time in the future that casting later would just push dots into
-		// TODO: This relies on a full set of invuln data ahead of time. Can this be trusted?
-		clip -= this.invuln.getInvulnerableUptime('all', event.timestamp, event.timestamp + this._statusDuration[statusId] + clip)
-
-		// Capping clip at 0 - less than that is downtime, which is handled by the checklist requirement
-		this._clip[statusId] = (this._clip[statusId] || 0) + Math.max(0, clip)
+		// Wait for when the status would typically drop without clipping - clipping a dot early isn't as problematic if it would
+		// just push it into invuln time.
+		this.addTimestampHook(
+			Math.min(
+				event.timestamp + this._statusDuration[statusId] + clip,
+				this.parser.eventTimeOffset + this.parser.pull.duration,
+			),
+			({timestamp}) => {
+				clip -= this.invuln.getInvulnerableUptime('all', event.timestamp, timestamp)
+
+				// Capping clip at 0 - less than that is downtime, which is handled by the checklist requirement
+				this._clip[statusId] = (this._clip[statusId] || 0) + Math.max(0, clip)
+			},
+		)
```

Comment:
core dots as well as a few old modules with similar logic: `this._clip` does not get _read_ outside `onComplete`, so timestamp hooking the invuln check timestamp (maxed at end of fight) allows us to get the same data as a look behind, rather than ahead.